src/SDKs/ManagementGroups/ManagementGroups.Tests/InMemoryTests/ManagementGroups.InMemoryTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Microsoft.Azure.Management.ManagementGroups;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest;
+using ResourceGroups.Tests;
+using Xunit;
+
+namespace Resource.Tests.InMemoryTests
+{
+    public class InMemoryManagementGroupsTests
+    {
+
+        public ManagementGroupsAPIClient GetManagementGroupsApiClient(RecordedDelegatingHandler handler)
+        {
+            var groupId = Guid.NewGuid().ToString();
+            var token = new TokenCredentials(groupId, "abc123");
+            handler.IsPassThrough = false;
+            var client = new ManagementGroupsAPIClient(token, handler);
+            HttpMockServer.Mode = HttpRecorderMode.Playback;
+            return client;
+        }
+
+        [Fact]
+        public void ListGroups()
+        {
+            var response = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent(@"{
```

Comment:
@rajshah11 why these responses hardcoded? why are they not getting recorded?

Reply:
These are InMemoryTests. All the scenarios are recorded here - [Scenario Tests](https://github.com/rajshah11/azure-sdk-for-net/blob/75490805e4f9742c6ac9ab60dc4e7118bbfb291c/src/SDKs/ManagementGroups/ManagementGroups.Tests/ScenarioTests/ManagementGroups.ScenarioTests.cs)